services/postprocess/src/xgboost_model/featurizer.py

```diff
@@ -0,0 +1,202 @@
+import ast
+
+expansion_delta = 50
+orig_size=1920
+
+def get_width_height(bbox):
+    return bbox[2]-bbox[0], bbox[3]-bbox[1]
+
+def compute_neighbors(center, predict_list):
+    center_bbox,_ = center
+    nbhds = []
+    for cand in predict_list:
+        if(center != cand):
+            nbhd_bbox,_ = cand
+            nbhd_bbox = [max(0, nbhd_bbox[0]-expansion_delta), max(0, nbhd_bbox[1]-expansion_delta), min(orig_size, nbhd_bbox[2]+expansion_delta), min(orig_size, nbhd_bbox[3]+expansion_delta)] 
+            iou = calculate_iou(nbhd_bbox, center_bbox)
+            if iou >0:
+                nbhds.append([cand, iou])
+    return nbhds
+
+def compute_neighbors_train(center, predict_list):
+    center_bbox,_,_,_,_ = center
+    nbhds = []
+    for cand in predict_list:
+        if(center != cand):
+            nbhd_bbox,_,_,_,_ = cand
+            nbhd_bbox = [max(0, nbhd_bbox[0]-expansion_delta), max(0, nbhd_bbox[1]-expansion_delta), min(orig_size, nbhd_bbox[2]+expansion_delta), min(orig_size, nbhd_bbox[3]+expansion_delta)] 
+            iou = calculate_iou(nbhd_bbox, center_bbox)
+            if iou >0:
+                nbhds.append([cand, iou])
+    return nbhds
+
+
+def not_ocr(text):
+    """
+    TODO: Docstring for not_ocr.
+
+    Args:
+        text (TODO): TODO
+
+    Returns: TODO
+
+    """
+    return 'ocr' not in text and 'rawtext' not in text and 'unicode' not in text
+
+def process_body(soup):
+    html_list = []
+    for seg_type in soup.find_all('div', not_ocr):
+        seg_class = " ".join(seg_type["class"])
+        hocr = seg_type.find_next('div', 'hocr')
+        if hocr is not None:
+            coordinates = hocr['data-coordinates']
+            spl = coordinates.split(' ')
+            spl = [int(x) for x in spl]
+            int_coords = [int(x) for x in spl]
+            score = hocr['data-score']
+            cls_scores = hocr['cls_scores']
+        #    if spl == ph:
+        #    seg_type["class"] = "Page Header"
+            lines = seg_type.find_all('span', 'ocr_line')
+            clean_line = ""
+            if len(lines) > 0:
+                line = lines[0]
+                clean_line = line.getText().strip().replace('\n', ' ').replace('  ', ' ').lower()
+            html_list.append((int_coords, cls_scores, score, seg_class, clean_line.encode('utf-8').strip()))
+    return html_list
+
+def process_html(html_file):
+    with codecs.open(html_file, "r", "utf-8") as fin:
+        soup = BeautifulSoup(fin, 'html.parser')
+        html_list = process_body(soup)    
+        return html_list
+
+def get_feat_vec(predict, predict_list, classes):
+    max_nhds = 15
+    feat_vec = []    
+    p_bb, p_cls_scores = predict
+    p_score, p_cls = p_cls_scores[0]
+    
+    # Neighbhorhood features
+    nbhds = compute_neighbors(predict, predict_list)
+    if len(nbhds) > max_nhds:
+        max_nhds = len(nbhds)
+    nbhds_sorted = sorted(nbhds, key=lambda nbhds: (nbhds[0]))
+    feat_nbhd1 = []
+    for nbhd in nbhds:
```

Comment:
I would change the name from nbhd to nbhr just because we're talking about neighbors, not other neighborhoods.